modules/apps/portal-store/portal-store-gcs-test/src/testIntegration/java/com/liferay/portal/store/gcs/test/GCSStoreStoreAreaProcessorTest.java

```diff
@@ -0,0 +1,248 @@
+			StringPool.QUESTION);
+
+		ConfigurationTestUtil.saveConfiguration(
+			_configuration,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"aes256Key", ""
+			).put(
+				"bucketName", "test"
+			).put(
+				"initialRetryDelay", "400"
+			).put(
+				"initialRPCTimeout", "120000"
+			).put(
+				"maxRetryAttempts", "5"
+			).put(
+				"maxRetryDelay", "10000"
+			).put(
+				"maxRPCTimeout", "600000"
+			).put(
+				"retryDelayMultiplier", "1.5"
+			).put(
+				"retryJitter", "false"
+			).put(
+				"rpcTimeoutMultiplier", "1.0"
+			).put(
+				"serviceAccountKey", ""
+			).build());
+	}
+
+	@AfterClass
+	public static void tearDownClass() throws Exception {
+		ConfigurationTestUtil.deleteConfiguration(_configuration);
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+	}
+
+	@Test
+	public void testCleanupDeleteStoreAreaKeepsRecentFiles() throws Exception {
+		String fileName = StringUtil.randomString();
+
+		StoreArea.withStoreArea(
+			StoreArea.DELETED,
+			() -> _store.addFile(
+				_group.getCompanyId(), _group.getGroupId(), fileName,
+				Store.VERSION_DEFAULT,
+				new UnsyncByteArrayInputStream(new byte[0])));
+
+		StoreArea.withStoreArea(
+			StoreArea.DELETED,
+			() -> Assert.assertTrue(
+				_store.hasFile(
+					_group.getCompanyId(), _group.getGroupId(), fileName,
+					Store.VERSION_DEFAULT)));
+
+		StoreAreaProcessor storeAreaProcessor = (StoreAreaProcessor)_store;
+
+		storeAreaProcessor.cleanUpDeletedStoreArea(
+			_group.getCompanyId(), 1, Duration.ofDays(1), StringPool.BLANK);
+
+		StoreArea.withStoreArea(
+			StoreArea.DELETED,
+			() -> Assert.assertTrue(
+				_store.hasFile(
+					_group.getCompanyId(), _group.getGroupId(), fileName,
+					Store.VERSION_DEFAULT)));
+	}
+
+	@Test
+	public void testCleanupDeleteStoreAreaRemovesFilesIncrementally()
+		throws Exception {
+
+		StoreArea.withStoreArea(
+			StoreArea.DELETED,
+			() -> {
+				String[] fileNames = new String[4];
+
+				for (int i = 0; i < fileNames.length; i++) {
+					fileNames[i] = StringUtil.randomString();
+				}
+
+				for (String fileName : fileNames) {
+					_store.addFile(
+						_group.getCompanyId(), _group.getGroupId(), fileName,
+						Store.VERSION_DEFAULT,
+						new UnsyncByteArrayInputStream(new byte[0]));
+				}
+			});
+
+		StoreAreaProcessor storeAreaProcessor = (StoreAreaProcessor)_store;
+
+		int runCount = 0;
+
+		String startOffset = StringPool.BLANK;
+
+		do {
+			startOffset = storeAreaProcessor.cleanUpDeletedStoreArea(
+				_group.getCompanyId(), 1, Duration.ofDays(-1), startOffset);
+
+			runCount++;
+		}
+		while (Validator.isNotNull(startOffset));
+
+		Assert.assertTrue(runCount > 1);
+
+		StoreArea.withStoreArea(
+			StoreArea.DELETED,
+			() -> {
+				String[] fileNames = _store.getFileNames(
+					_group.getCompanyId(), _group.getGroupId(),
+					StringPool.BLANK);
+
+				Assert.assertEquals(
+					Arrays.toString(fileNames), 0, fileNames.length);
+			});
```

Comment:
That would make the test brittle if we later change the parameters of the job. We don't really care about how many files are deleted on each run; this test is about verifying that deletion is incremental, and that when the processor says it's done (startOffset == ""), then there are no files (this is specific to this test; in reality this may not even happen).